src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Handlers/ElasticsearchContentIndexProfileHandler.cs

```diff
@@ -101,37 +101,28 @@ private static void PopulateTypeMapping(TypeMapping mapping, bool storeSource)
 
         mapping.DynamicTemplates ??= [];
 
-        if (!mapping.DynamicTemplates.Any(x => x.ContainsKey(_inheritedPostfixPattern)))
+        if (!mapping.DynamicTemplates.Any(x => x.Key == _inheritedPostfixPattern))
         {
             var inheritedPostfix = DynamicTemplate.Mapping(new KeywordProperty());
             inheritedPostfix.PathMatch = [_inheritedPostfixPattern];
             inheritedPostfix.MatchMappingType = ["string"];
-            mapping.DynamicTemplates.Add(new Dictionary<string, DynamicTemplate>()
-            {
-                { _inheritedPostfixPattern, inheritedPostfix },
-            });
+            mapping.DynamicTemplates.Add(new KeyValuePair<string, DynamicTemplate>(_inheritedPostfixPattern, inheritedPostfix));
         }
 
-        if (!mapping.DynamicTemplates.Any(x => x.ContainsKey(_idsPostfixPattern)))
+        if (!mapping.DynamicTemplates.Any(x => x.Key == _idsPostfixPattern))
```

Comment:
Please use string.Equals with StringComparar.Ordinal

Reply:
The string equality operator already uses ordinal string comparison: https://learn.microsoft.com/en-us/dotnet/csharp/how-to/compare-strings#default-ordinal-comparisons.

Reply:
Yes. The new analyzers will complain at using == for comparing strings.

Reply:
Do you really think we'd benefit from changing all `==` operators in order to "make it easier to search code for a certain string interpretation"? I don't see how. Would just add a bunch of trivial repetition all across the codebase.

Reply:
Updated, hate it :D.